django/contrib/postgres/operations.py

```diff
@@ -164,3 +164,65 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
             to_model_state = to_state.models[app_label, self.model_name_lower]
             index = to_model_state.get_index_by_name(self.name)
             schema_editor.add_index(model, index, concurrently=True)
+
+
+class CreateCollation(Operation):
```

Comment:
I think we should add also `RemoveCollation()` to fully support managing collations via migrations.

Reply:
I think it's a good idea, but as far as I can tell it wouldn't be reversible as we don't store the state. Outside of `RunPython` and `RunSQL`, are there any other operations that can't be reversed? I don't know of any. I'm not sure I'd like something provided by Django to not let me roll back migrations.

Maybe there's a way to reverse it that I'm missing, though.

Reply:
We can expose the same list of arguments (`name`, `locale`, `provider`, and `deterministic`) to both operations with a note in docs, that without providing them `RemoveCollation()` wouldn't properly recreate a collation on reverse.

Reply:
I thought about this but it seems quite verbose and prone to user-error. I suppose it's better than nothing, though, I'll try it out.